pkg/builder/local_build_executor.go

```diff
@@ -25,6 +25,37 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+type outputDirectory struct {
+	path string
+}
+
+func newOutputDirectory(workingDirectory string, pathDir string) outputDirectory {
```

Comment:
Isn't all of this the same as simply calling `path.Join(workingDirectory, pathDir)`

Reply:
Won't that mean that if an outputDir is specified as `../../../foo` and workingDirectory is `bar` then the output will be `bar/../../../foo`?

Because of the way that directory traversal works in the `filesystem` package, then we'll fall foul of https://github.com/buildbarn/bb-storage/blob/master/pkg/filesystem/local_directory.go#L21 ?

I see the value of keeping that validation, so there's no possibility of entering a directory above the build directory.

Reply:
`path.Join()` already does a `path.Clean()` for you. this means that in your example, the resulting path will be `../../foo`. See: https://play.golang.org/p/2rjqZOac11q

There is no need to have any additional logic, as `createOutputParentDirectory()` will still detect build directory escapes.

Reply:
Occam's razor for the win. Thanks @EdSchouten